### PR TITLE
BGP detailed  description for extended community and changes to add ISIS Route Range x-constraint in tx transmit choices for device traffic

### PR DESCRIPTION
**Pull request by SouravSinhaRoy**
Changes added to bgp extended community description  to add the valid combinations of types and sub types and also describe the format of the value field for each of the combinations. Also added isis names to Flow router container. The issue link is https://github.com/open-traffic-generator/models/issues/293

### CODE REVIEW

File: device/bgp/bgprouterange.yaml [hunk 1]
```diff
@@ -27,6 +27,25 @@ components:
         ext_communities:
           description: >-
             Optional Extended Community settings.
+            The Extended Communities Attribute is a transitive optional BGP attribute, with the Type Code 16. Community and Extended Communities  attributes
+            to trigger routing decisions, such as acceptance, rejection,  preference, or redistribution. An extended community is an 8-Bytes value.
```
Comment (apratimmukherjee) [inline device/bgp/bgprouterange.yaml:31]: are utilized to trigger ... ?

File: device/bgp/bgprouterange.yaml [hunk 2]
```diff
@@ -27,6 +27,25 @@ components:
...
+            It is also divided into two main parts. The first 2 Bytes of the community encode a type and sub-type fields and the last 6 Bytes carry a unique set
+            of data in a format defined by the type and sub-type field. Extended communities provide a larger  range for grouping or categorizing communities.
+            When type is administrator_as_2octet or administrator_as_4octet, the valid sub types are route target and origin. The valid value for 
+            administrator_as_2octet and administrator_as_4octet type is either two byte AS followed by four byte local administrator id or four byte AS followed by two 
+            byte local administrator id. 
+            When type is administrator_ipv4_address the valid sub types are route target and origin. The valid value for 
+            administrator_ipv4_address is an four byte IPv4 address followed by two byte local administrator id.
```
Comment (apratimmukherjee) [inline device/bgp/bgprouterange.yaml:38]: is a four byte ...

Comment (apratimmukherjee) [inline device/bgp/bgprouterange.yaml:38]: by a two byte

File: device/bgp/bgprouterange.yaml [hunk 3]
```diff
@@ -27,6 +27,25 @@ components:
...
+            administrator_as_2octet and administrator_as_4octet type is either two byte AS followed by four byte local administrator id or four byte AS followed by two 
+            byte local administrator id. 
+            When type is administrator_ipv4_address the valid sub types are route target and origin. The valid value for 
+            administrator_ipv4_address is an four byte IPv4 address followed by two byte local administrator id. 
+            When type is opaque, valid sub types are color and encapsulation. When sub type is color, first two bytes of the value field contain flags and last four bytes 
+            contains the value of the color. When sub type is encapsulation the first four bytes of value field are reserved and last two bytes carries the tunnel type from 
+            IANA's "ETHER TYPES" registry e.g IPv4 (protocol type = 0x0800), IPv6 (protocol type = 0x86dd), and MPLS (protocol type = 0x8847).
+            When type is administrator_as_2octet_link_bandwidth the valid sub type will be extended_bandwidth. The first two bytes of the value
```
Comment (apratimmukherjee) [inline device/bgp/bgprouterange.yaml:42]: ...sub type is extended_bandwidth.

File: device/bgp/bgprouterange.yaml [hunk 4]
```diff
@@ -27,6 +27,25 @@ components:
...
+            field will carry the AS and the last four bytes the bandwidth in IEEE floating point format.
```
Comment (apratimmukherjee) [inline device/bgp/bgprouterange.yaml:43]: contains the AS number ..... four bytes contain the bandwidth...

File: device/bgp/bgprouterange.yaml [hunk 5]
```diff
@@ -27,6 +27,25 @@ components:
...
+            contains the value of the color. When sub type is encapsulation the first four bytes of value field are reserved and last two bytes carries the tunnel type from 
+            IANA's "ETHER TYPES" registry e.g IPv4 (protocol type = 0x0800), IPv6 (protocol type = 0x86dd), and MPLS (protocol type = 0x8847).
+            When type is administrator_as_2octet_link_bandwidth the valid sub type will be extended_bandwidth. The first two bytes of the value
+            field will carry the AS and the last four bytes the bandwidth in IEEE floating point format. 
+            When type is evpn the valid subtype is mac_address. In the value field the low-order bit of the first byte(Flags) is defined as
+            the "Sticky/static" flag and may be set to 1, indicating the MAC address is static and cannot move. The second byte is reserved and the 
+            last four bytes contain the sequence number which is used to ensure that PEs retain the correct MAC/IP Advertisement route when multiple updates 
+            occur for the same MAC address. 
+            Note evpn type is defined mainly for use with evpn route updates and not for IPv4 and IPv6 route updates.
```
Comment (apratimmukherjee) [inline device/bgp/bgprouterange.yaml:48]: Once we have an agreement on the main body after fixing typos/nits .. same should be copied to all other individual uses for Extended Community in other BGP modules such as Policy and EVPN

**Review by apratimmukherjee — COMMENTED** [review]
Also did not find changes for route constraint. Maybe do in a different PR or add changes.?

File: device/bgp/bgprouterange.yaml (hunk 1 above)
Comment (SouravSinhaRoy) [reply device/bgp/bgprouterange.yaml]: changed

File: device/bgp/bgprouterange.yaml (hunk 2 above)
Comment (SouravSinhaRoy) [reply device/bgp/bgprouterange.yaml]: changed

Comment (SouravSinhaRoy) [reply device/bgp/bgprouterange.yaml]: changed

File: device/bgp/bgprouterange.yaml (hunk 3 above)
Comment (SouravSinhaRoy) [reply device/bgp/bgprouterange.yaml]: changed

File: device/bgp/bgprouterange.yaml (hunk 4 above)
Comment (SouravSinhaRoy) [reply device/bgp/bgprouterange.yaml]: changed

File: device/bgp/bgprouterange.yaml (hunk 5 above)
Comment (SouravSinhaRoy) [reply device/bgp/bgprouterange.yaml:48]: added in other places

**Review by SuryyaKrJana — APPROVED** [review]
looks fine

File: device/bgp/bgpsrtev4policy.yaml [hunk 6]
```diff
@@ -101,6 +101,24 @@ components:
         ext_communities:
           description: >-
             Optional Extended Community settings.
+            The Extended Communities Attribute is a transitive optional BGP attribute, with the Type Code 16. Community and Extended Communities  attributes
+            are utilized to trigger routing decisions, such as acceptance, rejection,  preference, or redistribution. An extended community is an 8-Bytes value.
+            It is also divided into two main parts. The first 2 Bytes of the community encode a type and sub-type fields and the last 6 Bytes carry a unique set
```
Comment (apratimmukherjee) [inline device/bgp/bgpsrtev4policy.yaml:106]: It is also divided ... -> It is divided ..

**Review by apratimmukherjee — APPROVED** [review]
One new comment left to address. Rest looks good to me.

File: device/bgp/bgpsrtev4policy.yaml (hunk 6 above)
Comment (SouravSinhaRoy) [reply device/bgp/bgpsrtev4policy.yaml]: updated

**Pull request merged**